x-pack/plugins/cloud_security_posture/public/pages/rules/rules_table.tsx

```diff
@@ -0,0 +1,136 @@
+};
+
+export const RulesTable = ({
+  toggleRule,
+  setSelectedRules,
+  setPagination,
+  perPage,
+  page,
+  ...props
+}: RulesTableProps) => {
+  const columns = useMemo(() => getColumns({ toggleRule }), [toggleRule]);
+
+  const items = useMemo(
+    () => (props.status === 'success' ? props.data?.slice() || [] : []),
+    [props.data, props.status]
+  );
+
+  const euiPagination: EuiBasicTableProps<RuleSavedObject>['pagination'] = {
+    pageIndex: Math.max(page - 1, 0),
+    pageSize: perPage,
+    totalItemCount: props.status === 'success' ? props.total : 0,
+    pageSizeOptions: [1, 5, 10, 25],
+    hidePerPageOptions: false,
+  };
+
+  const selection: EuiBasicTableProps<RuleSavedObject>['selection'] = {
+    selectable: () => true,
+    onSelectionChange: setSelectedRules,
+  };
+
+  const onTableChange = ({ page: _page }: Criteria<RuleSavedObject>) => {
+    if (!_page) return;
+    setPagination({ page: _page.index + 1, perPage: _page.size });
+  };
+
+  // Show "zero state"
+  if (props.status === 'success' && !props.data)
+    // TODO: use our own logo
+    return <EuiEmptyPrompt iconType="logoKibana" title={<h2>{TEXT.MISSING_RULES}</h2>} />;
+
+  return (
+    <EuiBasicTable
+      data-test-subj={TEST_SUBJECTS.CSP_RULES_TABLE}
+      loading={props.status === 'loading'}
+      error={props.error || undefined}
+      items={items}
+      columns={columns}
+      pagination={euiPagination}
+      onChange={onTableChange}
+      isSelectable={true}
+      selection={selection}
+      itemId={(v) => v.id}
+    />
+  );
+};
+
+const ruleNameRenderer = (name: string) => (
+  <EuiLink className="eui-textTruncate" title={name}>
+    {name}
+  </EuiLink>
+);
+
+// const timestampRenderer = (timestamp: string) =>
+//   moment.duration(moment().diff(timestamp)).humanize();
+
+interface GetColumnProps {
+  toggleRule: (rule: RuleSavedObject) => void;
+}
+
+const createRuleEnabledSwitchRenderer =
+  ({ toggleRule }: GetColumnProps) =>
+  (value: any, rule: RuleSavedObject) =>
+    (
+      <EuiSwitch
+        label=""
+        checked={value}
+        onChange={() => toggleRule(rule)}
+        data-test-subj={TEST_SUBJECTS.getCspRulesTableItemSwitchTestId(rule.attributes.id)}
+      />
+    );
+
+const getColumns = ({
+  toggleRule,
+}: GetColumnProps): Array<EuiTableFieldDataColumnType<RuleSavedObject>> => [
+  {
+    field: 'attributes.name',
+    name: TEXT.RULE_NAME,
+    width: '60%',
+    truncateText: true,
+    render: ruleNameRenderer,
+  },
+  {
+    field: 'section', // TODO: what field is this?
+    name: TEXT.SECTION,
+    width: '15%',
+  },
+  {
+    // TODO: get timestamp value
+    // add SavedObject["updated_at"] to SimpleSavedObject?
+    field: 'updated_at',
+    name: TEXT.UPDATED_AT,
+    width: '15%',
+    // render: timestampRenderer,
+  },
+  {
+    field: 'attributes.enabled',
+    name: TEXT.ENABLED,
+    render: createRuleEnabledSwitchRenderer({ toggleRule }),
```

Comment:
```suggestion
    render: (value: any, rule: RuleSavedObject) => 
    <EuiSwitch
        label=""
        checked={value}
        onChange={() => toggleRule(rule)}
        data-test-subj={TEST_SUBJECTS.getCspRulesTableItemSwitchTestId(rule.attributes.id)}
      />
```

nit: i think this is clearer.. i would treat `ruleNameRenderer` the same.. makes the `getColumns` function easier to visualize only by reading

Reply:
to me this suggestion is less clear, as it breaks the rest of the columns structure. each one takes a renderer function, so having just 1 of them do smth else feels wrong and inconsistent.

Reply:
```
const getColumns = ({
  toggleRule,
}: GetColumnProps): Array<EuiTableFieldDataColumnType<RuleSavedObject>> => [
  {
    field: 'attributes.name',
    name: TEXT.RULE_NAME,
    width: '60%',
    truncateText: true,
    render: (name: string) => (
      <EuiLink className="eui-textTruncate" title={name}>
        {name}
      </EuiLink>
    ),
  },
  {
    field: 'section', // TODO: what field is this?
    name: TEXT.SECTION,
    width: '15%',
  },
  {
    // TODO: get timestamp value
    // add SavedObject["updated_at"] to SimpleSavedObject?
    field: 'updated_at',
    width: '15%',
    name: TEXT.UPDATED_AT,
    render: (timestamp: string) => moment.duration(moment().diff(timestamp)).humanize(),
  },
  {
    field: 'attributes.enabled',
    name: TEXT.ENABLED,
    width: '10%',
    render: (value: any, rule: RuleSavedObject) => (
      <EuiSwitch
        label=""
        checked={value}
        onChange={() => toggleRule(rule)}
        data-test-subj={TEST_SUBJECTS.getCspRulesTableItemSwitchTestId(rule.attributes.id)}
      />
    ),
  },
];
```

of course, I meant that this should be done to all columns. just by looking at the example above, I can visualize this in my head already. no need to jump between 3 functions and no curry function.

still of course if you dont like it don't do it. just wanted to be clear that I indeed meant this treatment for all the columns